packages/web/src/components/list/MultiDataList.js

```diff
@@ -364,28 +364,53 @@ class MultiDataList extends Component {
 
 	handleClick = (e) => {
 		const { value, onChange } = this.props;
-		const { value: listValue } = e.target;
-		if (value === undefined) {
-			this.setValue(listValue);
-		} else if (onChange) {
-			onChange(listValue);
+		if (isEvent(e)) {
+			const { value: listValue } = e.target;
+			if (value === undefined) {
+				this.setValue(listValue);
+			} else if (onChange) {
+				onChange(listValue);
+			}
+		} else {
+			this.setValue(e);
 		}
 	};
 
-	render() {
-		const { selectAllLabel, showCount, renderListItem } = this.props;
-		const { options } = this.state;
+	getComponent() {
+		const { currentValue } = this.state;
+		const data = {
+			value: currentValue,
+			data: this.listItems,
+			handleChange: this.handleClick,
+		};
+		return getComponent(data, this.props);
+	}
 
-		if (options.length === 0) {
-			return null;
-		}
+	get hasCustomRenderer() {
+		return hasCustomRenderer(this.props);
+	}
+
+	get listItems() {
+		const { options } = this.state;
 
 		const listItems = options.filter((item) => {
 			if (this.props.showSearch && this.state.searchTerm) {
 				return item.label.toLowerCase().includes(this.state.searchTerm.toLowerCase());
 			}
 			return true;
 		});
+		return listItems;
+	}
+
+	render() {
+		const { selectAllLabel, showCount, renderItem } = this.props;
+		const { options } = this.state;
+
+		if (!this.hasCustomRenderer && options.length === 0) {
```

Comment:
This should be `||` (OR) imo

Reply:
`&&` is intentional, User should has control to decide that what to do with empty results when custom renderer is being used.

Reply:
got it! 👍